libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -142,14 +145,32 @@ class MapboxRouteLineView(options: MapboxRouteLineOptions) {
      * the layers if they have not yet been initialized. If you have a use case for initializing
      * the layers in advance of any API calls this method may be used.
      *
+     * If the layers already exist they will be removed and re-initialized with the options provided.
+     *
      * Each [Layer] added to the map by this class is a persistent layer - it will survive style changes.
      * This means that if the data has not changed, it does not have to be manually redrawn after a style change.
      * See [Style.addPersistentStyleLayer].
      *
      * @param style a valid [Style] instance
      */
     fun initializeLayers(style: Style) {
-        MapboxRouteLineUtils.initializeLayers(style, options)
+        resetLayers(style)
+    }
+
+    /**
+     * Updates the [MapboxRouteLineOptions] used for the route line related layers and initializes the layers.
+     * If the layers already exist they will be removed and re-initialized with the options provided.
+     *
+     * Each [Layer] added to the map by this class is a persistent layer - it will survive style changes.
+     * This means that if the data has not changed, it does not have to be manually redrawn after a style change.
+     * See [Style.addPersistentStyleLayer].
+     *
+     * @param style a valid [Style] instance
+     * @param options used for the route line related layers.
+     */
+    fun initializeLayers(style: Style, options: MapboxRouteLineOptions) {
```

Comment:
Let's please remove this function for now and treat options as immutable. The mutation can be achieved by recreating the instance and that's enough to satisfy the original request.

If we want to add a mutation function, let's consider is separately.

Reply:
I don't understand the motivation for this.  This method provides a better developer experience. Why treat the options as immutable when then are mutable.  It doesn't make sense to me to force a developer into an implicit reinitialization of the layers by 1. creating a new instance 2. calling a render method, when it costs nothing to provide an explicit single step via this method.

Reply:
I don't understand the motivation for this.  This method provides a better developer experience. Why treat the options as immutable when then are mutable.  It doesn't make sense to me to force a developer into an implicit reinitialization of the layers by 1. creating a new instance 2. calling a render method, when it costs nothing to provide an explicit single step via this method.

Reply:
Calling this function would force developers to call `render` again anyway because we're clearing all the data. I'd like us to avoid expanding route line APIs any further beyond what's necessary to satisfy a requirement because they've been growing unnecessarily for a long time.